lib/gateway/Shard.js

```diff
@@ -2070,6 +2053,204 @@ class Shard extends EventEmitter {
+                        flags: member.flags
+                    };
+                }
+                member = channel.members.update(packet.d, this.client);
+                /**
+                * Fired when a thread member is updated
+                * @event Client#threadMemberUpdate
+                * @prop {NewsThreadChannel | PrivateThreadChannel | PublicThreadChannel} channel The channel
+                * @prop {ThreadMember} member The updated thread member
+                * @prop {Object} oldMember The old thread member data
+                * @prop {Number} oldMember.flags User thread settings
+                */
+                this.emit("threadMemberUpdate", channel, member, oldMember);
+                break;
+            }
+            case "THREAD_MEMBERS_UPDATE": {
+                const channel = this.client.getChannel(packet.d.id);
+                if(!channel) {
+                    this.emit("debug", `Missing channel ${packet.d.id} in THREAD_MEMBERS_UPDATE`);
+                    break;
+                }
+                channel.update(packet.d);
+                let removedMembers;
+                let addedMembers;
+                if(packet.d.removed_member_ids) {
+                    removedMembers = packet.d.removed_member_ids.map((id) => channel.members.remove({id}) || {id});
+                }
+                if(packet.d.added_members) {
+                    addedMembers = packet.d.added_members.map((m) => channel.members.update(m, this.client));
+                }
+                /**
+                * Fired when anyone is added or removed from a thread. If the `guildMembers` intent is not specified, this will only apply for the current user
+                * @event Client#threadMembersUpdate
+                * @prop {NewsThreadChannel | PrivateThreadChannel | PublicThreadChannel} channel The thread channel
+                * @prop {Array<ThreadMember | Object>} removedMembers An array of members that were removed from the thread channel. If a member is uncached, it will be an object with an `id` key. No other property is guaranteed
+                * @prop {Array<ThreadMember>} addedMembers An array of members that were added to the thread channel
+                */
+                this.emit("threadMembersUpdate", channel, removedMembers, addedMembers);
+                break;
+            }
+            case "STAGE_INSTANCE_CREATE": {
+                const guild = this.client.guilds.get(packet.d.guild_id);
+                if(!guild) {
+                    this.emit("debug", `Missing guild ${packet.d.guild_id} in STAGE_INSTANCE_CREATE`);
+                    break;
+                }
+                /**
+                * Fired when a stage instance is created
+                * @event Client#stageInstanceCreate
+                * @prop {StageInstance} stageInstance The stage instance
+                */
+                this.emit("stageInstanceCreate", guild.stageInstances.add(packet.d));
```

Comment:
StageInstance requires a client property which is not provided when adding a new stage instance to the guild. Which causes an error: `TypeError: Cannot read property 'getChannel' of undefined at new StageInstance` [here](https://github.com/bsian03/eris/blob/d3043f36708fd48ef512faf473ad8768c45d2efb/lib/structures/StageInstance.js#L18)

Reply:
sad